# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

___

### **Hipotesis:**

**Hipotesis Nula(𝐻0)**:
- No hay diferencia en la pérdida de peso media: **𝜇1=𝜇2**

**Hipotesis Alternativa (𝐻1)**:
- Sí hay diferencia: **𝜇1≠𝜇2**


In [1]:
import numpy as np
from scipy import stats

diet1 = np.array([2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5])
diet2 = np.array([3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2])

#Usamos una prueba t para muestras independientes (son dos grupos distintos de personas).

t_stat, p_value = stats.ttest_ind(diet1, diet2, equal_var=True)
print("t =", t_stat)
print("p-valor =", p_value)


t = -3.5383407969933938
p-valor = 0.004083270191713912


**Conclusion**:

Como 𝑝=0.0041<0.05:
- Rechazamos la hipotesis Nula (𝐻0)
- Concluimos que sí existe una diferencia significativa en la pérdida de peso media entre las dos dietas.

Además, como la media de la dieta 2 es mayor, podemos decir que, con estos datos, la dieta 2 parece producir más pérdida de peso que la dieta 1.

___

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

___

### **Datos**

In [2]:
fert1 = np.array([20, 21, 20, 19, 20])
fert2 = np.array([22, 21, 23, 22, 21])
fert3 = np.array([24, 23, 22, 23, 24])

### **Hipótesis**

**Hipotesis Nula (𝐻0)**: 
- Los tres fertilizantes tienen el mismo rendimiento medio

𝜇1=𝜇2=𝜇3

**Hipotesis Alternativa (𝐻1)**:
- Al menos uno de los fertilizantes tiene un rendimiento medio diferente
(no todos los μ son iguales)

In [3]:
F, p = stats.f_oneway(fert1, fert2, fert3)
print("F =", F)
print("p-valor =", p)

#Cual es mejor?
print("Media fert1:", fert1.mean())
print("Media fert2:", fert2.mean())
print("Media fert3:", fert3.mean())


F = 20.31578947368421
p-valor = 0.000140478247931904
Media fert1: 20.0
Media fert2: 21.8
Media fert3: 23.2


**Observacion**:

- 𝐹≈20.32
- 𝑝≈0.00014

Con un nivel de significancia 𝛼=0.05:

Como 𝑝=0.00014<0.05, rechazamos la **Hipotesis Nula (𝐻0)**.

- Hay diferencia significativa global (ANOVA).

- El fertilizante 3 es el que presenta mayor rendimiento promedio.

- Podemos confirmar qué pares son diferentes aplicando pruebas post-hoc (por ejemplo, Tukey o t-tests por pares).